docs/types/value-types.rst

```diff
@@ -300,6 +300,16 @@ All three functions ``call``, ``delegatecall`` and ``staticcall`` are very low-l
 The ``gas`` option is available on all three methods, while the ``value`` option is not
 supported for ``delegatecall``.
 
+.. note::
+    ``Gas counting`` also applies to ``staticcall``. This is true for both reading state during contract execution
```

Comment:
This sounds more like a comment on `web3.call` than on `address.staticcall` to me. If at all, maybe the `view` modifier should be the place to put it.

Reply:
```
(bool success,  bytes memory returndata) 
     = address(_condition).staticcall.gas(_conditionGasActionGasMinExecutionGas[0])(
           _conditionPayloadWithSelector
 );
```

Some background.
In our contract we have a `view` function that wraps another `view` function with a `low-level` `staticcall` and that limits the `gas` sent with the `staticcall`.  This function is meant to be called off-chain first and then, if the `staticcall` returns true, the caller will send a transaction to another function on the same contract on-chain during whose execution the same `view` wrapper will be called again. 

The reason why we `gas limited` the `staticcall` was to regulate gas consumption and cost on-chain during the second step transaction execution.  Fo us the unexpected byproduct of this hardcoded staticcall gas stipend limit was that the view function sometimes did not return true or false when we called it from Javascript in the first step, but instead bubbled up an error message, the cause of which we could locate to the `gas limit` passed with the `staticcall`. 

So here is why I think that this is more relevant for  a note on `staticcall` and not `web3.call` or `ethers.js` (which we use): as far as I am aware you do not normally pass a `gas` `override` along with a web3 call or an ethers.js call. And even if you did, we would have no way of affecting the described behaviour, since the `gas limit` passed with the `staticcall` is immutably coded into the deployed smart contract instance. The problem is not that there is not enough `gasleft()` when the `staticcall.gas(stipend)` is made, but rather that the `gas` stipend for the `staticcall` is apparently fully consumed during the `staticcall`.

Reply:
Maybe we should just warn that in general, relying on anything to do with gas is usually a bad idea, has many pitfalls, and access to gas might change in the future.

Reply:
@gitpusha would you like to update the PR?

Reply:
Hi @leonardoalt. Sure. What exactly do you want me to update? Make it more general as Chris suggested?

Reply:
@gitpusha That'd be great!

Reply:
@leonardoalt done. The formatting looks a bit off but it looked fine in the Preview/rich diff.